lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -991,6 +996,276 @@ def _generate_report(self, trans, invocation_id, **kwd):
+                        software_prerequisites.append(software_prerequisite)
+                        tools.append(software_prerequisite['uri']['uri'])
+                except Exception:
+                    continue
+
+            if step.workflow_step.type == 'data_input':
+                for output_assoc in step.output_datasets:
+                    encoded_dataset_id = trans.security.encode_id(output_assoc.dataset.id)
+                input_obj = {
+                    'filename': step.workflow_step.label,
+                    'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                    'access_time': step.workflow_step.update_time.isoformat(),
+                }
+                input_subdomain.append(input_obj)
+
+            if step.workflow_step.type == 'data_collection_input':
+                encoded_dataset_id = trans.security.encode_id(step.workflow_step.id)
+                input_obj = {
+                    'filename': step.workflow_step.label,
+                    'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                    'access_time': step.workflow_step.update_time.isoformat(),
+                }
+                input_subdomain.append(input_obj)
+
+        usability_domain = []
+        for a in stored_workflow.annotations:
+            usability_domain.append(a.annotation)
+        for h in history.annotations:
+            usability_domain.append(h.annotation)
+
+        parametric_domain = []
+        for inv_step in workflow_invocation.steps:
+            try:
+                for k, v in inv_step.workflow_step.tool_inputs.items():
+                    param, value, step = k, v, inv_step.workflow_step.order_index
+                    parametric_domain.append({'param': param, 'value': value, 'step': step})
+            except Exception:
+                continue
+
+        execution_domain = {
+            'script_access_type': 'a_galaxy_workflow',
+            'script': [url_for('workflows', encoded_workflow_id=encoded_workflow_id)],
+            'script_driver': 'Galaxy',
+            'software_prerequisites': software_prerequisites,
+            'external_data_endpoints': [
+                {
+                    'name': 'Access to Galaxy',
+                    'url': url_for('/', qualified=True)
+                },
+                kwd.get('external_data_endpoints')
+            ],
+            'environment_variables': kwd.get('environment_variables', {})
+        }
+
+        extension = [
+            {
+                'extension_schema': 'https://raw.githubusercontent.com/biocompute-objects/extension_domain/6d2cd8482e6075746984662edcf78b57d3d38065/galaxy/galaxy_extension.json',
+                'galaxy_extension': {
+                    'galaxy_url': url_for('/', qualified=True),
+                    'galaxy_version': VERSION,
+                    # 'aws_estimate': aws_estimate,
```

Comment:
These seem like they are serving as fine TODOs to me - there is already metrics being collected above and there is an open PR to make progress on aws_estimate I believe. What if I simply added the word TODO before them - then it would be like @jmchilton code 😅